docs/modules/ROOT/pages/references/quality-requirements/maintainability/portal-api-modularity.adoc

```diff
@@ -0,0 +1,23 @@
+= Portal API Modularity
+
+Source::
+Portal API
+
+Stimulus::
+Portal API connects to external backend system
+
+Environment::
+Portal API
+
+Artifact::
+Portal API and external backend system
+
+Response::
+Data from the external can be read and written, independent of the exact external system implementation, as long as it serves the same purpose
+
+Response measure::
+Data presented by the API is exactly the same, no matter what the backend system implementation looks like
+
+Rationale::
+The Portal API should provide modularity so that the external system which the API works with can be exchanged, as long as the external system serves the same purpose.
+For example the ERP which contains invoices and billing information can be exchanged by changing the backend connection implementation, the frontend API presented by the Portal API stays the same and doesn't change.
```

Comment:
If you intended to document the API with a plugin-like system to support multiple different providers, then the keyword is "Replaceability" within the Portability category. "Modularity" is something else ("minimal impact" != "no impact").

So this requirement goes into the Portability category IMO :)
```suggestion
= External components as plugins

Source::
{product} administrator

Stimulus::
Desire to integrate a third-party component

Environment::
Portal API

Artifact::
Portal API, third-party component

Response::
Data from the external can be read and written, independent of the exact external system implementation, as long as it serves the same purpose

Response measure::
Third-party system behaves according to the API specification (see https://en.wikipedia.org/wiki/Design_by_contract[Design by contract principle] applied to components)

Rationale::
The Portal API should provide replaceability so that the external system which the API works with can be exchanged, as long as the external system serves the same purpose.
For example an ERP as third-party component responsible for invoices can be exchanged with another implementation so that the Portal API stays the same and continues serving the clients without requiring changes.
```